### PR TITLE
feat: increased unit test coverage in the pkg/acl package

### PR DESCRIPTION
**Pull request by pawelpaszki**
## Description
feat: increased unit test coverage in the pkg/acl package

Before changes the coverage was:

```
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list.go:12:					IsUserDenied						100.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list.go:24:					NewAccessControlListConfig				0.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list.go:31:					AddFlags						0.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list.go:36:					ReadFiles						0.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list.go:45:					readDenyListConfigFile					0.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list_middleware.go:15:				NewAccessControlListMiddleware				100.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list_middleware.go:23:				Authorize						62.5%
```

After adding new tests the coverage is (addFlags is not really testable):

```
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list.go:12:					IsUserDenied						100.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list.go:24:					NewAccessControlListConfig				100.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list.go:31:					AddFlags						0.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list.go:36:					ReadFiles						100.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list.go:45:					readDenyListConfigFile					100.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list_middleware.go:15:				NewAccessControlListMiddleware				100.0%
github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/acl/access_control_list_middleware.go:23:				Authorize						87.5%
```

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has created for changes required on the client side

### CODE REVIEW

File: pkg/acl/access_control_list_test.go [hunk 1]
```diff
@@ -34,13 +34,97 @@ func Test_DeniedUsers_IsUserDenied(t *testing.T) {
...
+		},
+		{
+			name: "should not return an error when deny list is disabled",
+			args: args{
+				file:           "config/deny-list-configuration.yaml",
+				deniedUsers:    &DeniedUsers{},
+				enableDenyList: false,
+			},
+			wantErr: false,
+		},
```
Comment (machi1990) [inline pkg/acl/access_control_list_test.go:116]: There are missing test cases for when file path is not valid with deny list enabled/disabled?

File: pkg/acl/access_control_list_middleware_test.go [hunk 2]
```diff
@@ -48,22 +47,34 @@ func Test_AccessControlListMiddleware_UserHasNoAccess(t *testing.T) {
...
+			wantHttpStatus: http.StatusForbidden,
+		},
+		{
+			name: "returns 200 status if denyList is disabled",
+			arg: &acl.AccessControlListConfig{
+				EnableDenyList: false,
+			},
+			wantErr:        false,
+			wantHttpStatus: http.StatusOK,
 		},
```
Comment (machi1990) [inline pkg/acl/access_control_list_middleware_test.go:71]: We are missing a test case when user is allowed to access the service and deny list is enabled

File: pkg/acl/access_control_list_test.go (hunk 1 above)
Comment (pawelpaszki) [reply pkg/acl/access_control_list_test.go]: good point. added tests to address both comments

**Comment by machi1990** [comment]
Thanks @pawelpaszki, how are the numbers looking now after the additional of the test cases?

File: pkg/acl/access_control_list_test.go (hunk 1 above)
Comment (machi1990) [reply pkg/acl/access_control_list_test.go]: Thanks!

**Comment by pawelpaszki** [comment]
> Thanks @pawelpaszki, how are the numbers looking now after the additional of the test cases?

I've included the coverage percentage in the PR description

**Comment by machi1990** [comment]
> > Thanks @pawelpaszki, how are the numbers looking now after the additional of the test cases?
> 
> I've included the coverage percentage in the PR description

Oh yes. I've seen that. I was asking about the new coverage after the additional of the test cases following the review.

**Comment by pawelpaszki** [comment]
> > > Thanks @pawelpaszki, how are the numbers looking now after the additional of the test cases?
> > 
> > 
> > I've included the coverage percentage in the PR description
> 
> Oh yes. I've seen that. I was asking about the new coverage after the additional of the test cases following the review.

still the same :) due to implementation of tested features :)

File: pkg/acl/access_control_list_middleware_test.go [hunk 3]
```diff
@@ -95,20 +114,21 @@ func Test_AccessControlListMiddleware_UserHasNoAccess(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
```
Comment (ziccardi) [inline pkg/acl/access_control_list_middleware_test.go:116]: It is not part of this PR, but can we change this to 
```go
Expect(err).NotTo(BeOccurred())
```

File: pkg/acl/access_control_list_middleware_test.go [hunk 4]
```diff
@@ -95,20 +114,21 @@ func Test_AccessControlListMiddleware_UserHasNoAccess(t *testing.T) {
...
-			if err != nil {
-				t.Fatal(err)
+			Expect(rr.Code).To(Equal(tt.wantHttpStatus))
+
+			if tt.wantErr {
+				Expect(rr.Header().Get("Content-Type")).To(Equal("application/json"))
+				var data map[string]string
+				err = json.Unmarshal(body, &data)
+				if err != nil {
+					t.Fatal(err)
```
Comment (ziccardi) [inline pkg/acl/access_control_list_middleware_test.go:124]:
```suggestion
Expect(err).NotTo(BeOccurred())
```

**Comment by machi1990** [comment]
> > > > Thanks @pawelpaszki, how are the numbers looking now after the additional of the test cases?
> > > 
> > > 
> > > I've included the coverage percentage in the PR description
> > 
> > 
> > Oh yes. I've seen that. I was asking about the new coverage after the additional of the test cases following the review.
> 
> still the same :) due to implementation of tested features :)

Thanks

File: pkg/acl/access_control_list_middleware_test.go (hunk 3 above)
Comment (pawelpaszki) [reply pkg/acl/access_control_list_middleware_test.go]: changed that. thanks

**Comment by pawelpaszki** [comment]
@machi1990 @ziccardi - can you approve if you are happy with the changes please?

**Review by machi1990 — APPROVED** [review]
lgtm

Thanks Pawel
